### PR TITLE
Date format support

### PR DESCRIPTION
**Pull request by divanovGH**

### CODE REVIEW

File: voluptuous/tests/tests.py [hunk 1]
```diff
@@ -745,3 +745,13 @@ def test_named_tuples_validate_as_tuples():
     Schema((int, int))(t)
     Schema(NT(int, int))(nt)
     Schema(NT(int, int))(t)
+
+
```
Comment (tusharmakkar08) [inline voluptuous/tests/tests.py]: Can you add a test case where `DateInvalid` is catched ?

File: voluptuous/validators.py [hunk 2]
```diff
@@ -587,6 +587,27 @@ def __repr__(self):
...
+    """Validate that the value matches the date format."""
+
+    DEFAULT_FORMAT = '%Y-%m-%d'
+    FORMAT_DESCRIPTION = 'yyyy-mm-dd'
+
+    def __call__(self, v):
+        try:
+            datetime.datetime.strptime(v, self.format).date()
+            if len(v) != len(self.FORMAT_DESCRIPTION):
+                raise ValueError
```
Comment (tusharmakkar08) [inline voluptuous/validators.py]: ValueError doesn't provide sufficient information. Use DateInvalid with different string input.

File: voluptuous/tests/tests.py (hunk 1 above)
Comment (divanovGH) [inline voluptuous/tests/tests.py]: Yep! )

Comment (divanovGH) [inline voluptuous/tests/tests.py]: This test is already written in the 760 line

**Comment by tusharmakkar08** [comment]
This looks fine. Please squash your commits.

**Comment by divanovGH** [comment]
Done!

**Pull request merged**